### PR TITLE
Support multiple values per one cost filter in aws_budgets_budget resource

### LINKED ISSUE #5890: Budget with multiple TagKeyValue cost filters
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to create a Budget with more than one TagKeyValue in cost_filters.  AWS supports this, but AFAICT Terraform currently does not.

### New or Affected Resource(s)

* aws_budgets_budget

### Potential Terraform Configuration

```hcl
provider "aws" {
  region = "us-east-2"
}

resource "aws_budgets_budget" "test" {
  name              = "test"
  budget_type       = "COST"
  limit_amount      = "500"
  limit_unit        = "USD"
  time_unit         = "MONTHLY"
  time_period_start = "2018-01-01_00:00"

  cost_filters {
    TagKeyValue = ["user:Service$IP Address Management", "user:Environment$Dev"]
  }

  cost_types {
    use_amortized = true
  }
}
```

Currently (Terraform v0.11.8, provider.aws v1.36.0) this fails with
```
Error: aws_budgets_budget.test: cost_filters (TagKeyValue): '' expected type 'string', got unconvertible type '[]interface {}'
```

### etc

Just for fun, let's comment out the TagKeyValue line above and then `terraform apply` (creating a Budget with no cost_filters).  Now I can add my desired filters manually using the AWS web console.

At this point, we can see that AWS CLI returns an array for TagKeyValue (hence my belief that Terraform ought to accept an array for it):

```
$ aws budgets describe-budget --account-id 378517677616 --budget-name test
{
    "Budget": {
        "BudgetType": "COST",
        "TimePeriod": {
            "End": 3706473600.0,
            "Start": 1535760000.0
        },
        "CalculatedSpend": {
            "ActualSpend": {
                "Unit": "USD",
                "Amount": "0"
            }
        },
        "CostTypes": {
            "UseBlended": false,
            "IncludeCredit": true,
            "IncludeSubscription": true,
            "IncludeRecurring": true,
            "IncludeUpfront": true,
            "UseAmortized": true,
            "IncludeSupport": true,
            "IncludeDiscount": true,
            "IncludeRefund": true,
            "IncludeTax": true,
            "IncludeOtherSubscription": true
        },
        "BudgetName": "test",
        "BudgetLimit": {
            "Unit": "USD",
            "Amount": "500"
        },
        "TimeUnit": "MONTHLY",
        "CostFilters": {
            "TagKeyValue": [
                "user:Service$IP Address Management",
                "user:Environment$Dev"
            ]
        }
    }
}
```

### LINKED ISSUE #10692: aws_budgets_budget.limit_amount not settling due to differences in decimal places
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```shell
$ terraform -v
Terraform v0.12.12
+ provider.aws v2.33.0
+ provider.http v1.1.1
+ provider.kubernetes v1.9.0
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2
```

### Affected Resource(s)

* aws_budgets_budget

### Terraform Configuration Files

```hcl
resource "aws_budgets_budget" "budget" {
  name              = "Monthly Budget"
  budget_type       = "COST"
  time_period_start = "2019-09-01_00:00"
  time_unit         = "MONTHLY"
  limit_amount = 2000
  limit_unit   = "USD"

  cost_types {
    include_credit = false
    include_refund = false
  }

  notification {
    comparison_operator = "GREATER_THAN"
    threshold           = "100"
    threshold_type      = "PERCENTAGE"
    notification_type   = "ACTUAL"

    subscriber_email_addresses = [
      "ops@example.com",
    ]
  }
}
```

### Output

```

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_budgets_budget.budget will be updated in-place
  ~ resource "aws_budgets_budget" "budget" {
        account_id        = "608019992430"
        budget_type       = "COST"
        cost_filters      = {}
        id                = "608019992430:Monthly Budget"
      ~ limit_amount      = "2000.0" -> "2000"
        limit_unit        = "USD"
        name              = "Monthly Budget"

... snip ...

aws_budgets_budget.budget: Modifying... [id=608019992430:Monthly Budget]
2019/10/31 02:16:09 [WARN] Provider "aws" produced an unexpected new value for aws_budgets_budget.budget, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .limit_amount: was cty.StringVal("2000"), but now cty.StringVal("2000.0")
2019/10/31 02:16:09 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/10/31 02:16:09 [TRACE] <root>: eval: *terraform.EvalWriteState
2019/10/31 02:16:09 [TRACE] EvalWriteState: writing current state object for aws_budgets_budget.budget
aws_budgets_budget.budget: Modifications complete after 5s [id=608019992430:Monthly Budget]
```

### Expected Behavior

Applying the budget, then applying again should result in no changes

### Actual Behavior

Budget will not settle

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

N/A

## Work around

In my real terraform file I am loading the budget from a variable; so changing
```hcl
  limit_amount = var.my_budget
```
to
```hcl
  limit_amount = format("%.1f", var.my_budget)
```
allows the resource to settle.

### LINKED ISSUE #12669: Create was budget resource needs to support multiple linked accounts
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
The create budget resource needs to support multiple linked accounts. AWS budgets support using multiple linked accounts when creating a budget. This is especially important in a multi-account setup, so you can configure a budget across multiple accounts (dev, stage, prod) for a particular product or business unit. For some reason LinkedAccount in the C

### New or Affected Resource(s)

* aws_budgets_budget

### Potential Terraform Configuration

```hcl

resource "aws_budgets_budget" "product1_allservices" {
  name              = "budget-product1-allservices"
  budget_type       = "COST"
  limit_amount      = "50000"
  limit_unit        = "USD"
  time_unit         = "MONTHLY"
  time_period_start = "2020-03-01_00:00"

  cost_filters = {
    LinkedAccount = ["XXXX", "xxxx2"] //dev & prod accounts, currently this has to be a string
  }

  notification {
    comparison_operator        = "GREATER_THAN"
    threshold                  = 100
    threshold_type             = "PERCENTAGE"
    notification_type          = "FORECASTED"
    subscriber_email_addresses = ["a@gmail.com"]
  }
}


```

### References

* #0000

### LINKED ISSUE #13288: aws_budgets_budget create and update fail with multilpe TagKeyValue cost_filters
_This issue was originally opened by @ndench as hashicorp/terraform#24924. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._
    
<hr>
    
When managing an `aws_budgets_budget` resource, a single `cost_filter` for `TagKeyValue` is all that can be specified, if multiple are specified, an error occurs on both create and update.

The create/update must be applied manually in the AWS Budgets Console, and then `terraform apply` successfully completes as long as there are no required changes to the `aws_budgets_budget` resource.

### Terraform Version
```
Terraform v0.12.24
+ provider.aws v2.41.0
```

### Terraform Configuration Files
```hcl
resource "aws_budgets_budget" "test" {
  name = "test-budget"
  budget_type = "COST"
  limit_amount = "1.0"
  limit_unit = "USD"
  time_period_start = "2020-05-01_00:00"
  time_unit = "MONTHLY"

  cost_filters = {
    TagKeyValue = "user:App$my-app,user:App$my-app-2"
  }
}
```

### Expected Behavior
The budget should be created or updated.

### Actual Behavior
When trying to initially create the resource:
```
Error: create budget failed: InvalidParameterException: Failed to call CreateBudget for [AccountId: 771882897437] - null
        status code: 400, request id: b471c15f-dabb-4656-8267-bc2f4e49fa6e

  on main.tf line 525, in resource "aws_budgets_budget" "test":
 525: resource "aws_budgets_budget" "test" {
```

When trying to update a resource:
```
Error: update budget failed: InvalidParameterException: Failed to call UpdateBudget for [AccountId: 771882897437] - null
        status code: 400, request id: 99f2e1b0-0a5f-4ea6-8b23-c5bdacd42f24

  on main.tf line 525, in resource "aws_budgets_budget" "test":
 525: resource "aws_budgets_budget" "test" {

```

### Steps to Reproduce
1. Add `aws_budgets_budget` resource specified above
2. `terraform apply` -> fails with `update budget failed` error
4. Comment out the `TagKeyValue` line
5. `terraform apply` -> success
6. Uncomment the `TagKeyValue` line
7. `terraform apply` -> fails with `update budget failed` error
8. Add the specified tags in the AWS Budgets console
9. `terraform apply` -> success with no changes
10. Change some other parameter (not TagKeyValue)
11. `terraform apply` -> fails with `update budget failed` error
12. Remove one of the tags from `TagKeyValue`
13. `terraform apply` -> success

### Additional Context
The tags specified in `TagKeyValue` need to be [cost allocation tags](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/cost-alloc-tags.html), I've been using user defined ones which I had to enable in the Billing Console. 

It appears that the [aws budgets cli](https://docs.aws.amazon.com/cli/latest/reference/budgets/create-budget.html) takes `TagKeyValue` as a list, maybe the data type is incorrect?

### LINKED ISSUE #19803: aws_budgets_budget: time_period_start to be optional instead of required
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Will be useful to have the `time_period_start` argument as optional. If you don't specify a start date, the upstream API will default it to the start of your chosen time period (DAILY, MONTHLY, QUARTERLY, or ANNUALLY) from `time_unit`.

### New or Affected Resource(s)

* aws_budgets_budget

### References

* https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_budgets_Budget.html#awscostmanagement-Type-budgets_Budget-TimePeriod
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/budgets_budget

### PR DESCRIPTION
**Pull request by mozamimy**
Current aws_budgets_budget resource does not support multiple cost filter values because the `cost_filters` map takes only one string value per string key. However, AWS Budgets API and its console can handle multiple values per one cost filter key.

This change makes it to be able to handle multiple cost filter values like following syntax. This is a similar concept to `parameter` attribute of [Resource: aws_db_parameter_group](https://www.terraform.io/docs/providers/aws/r/db_parameter_group.html).

```hcl
resource "aws_budgets_budget" "monthly_cost_budget" {
  # :
  # (snip)
  # :

  cost_filter {
    name = "Region"
    values = [
      "ap-northeast-1",
      "us-east-1",
    ]
  }

  cost_filter {
    name = "PurchaseType"
    values = [
      "Spot",
    ]
  }

  # :
  # (snip)
  # :
}
```

I tried to implement like "Potential Terraform Configuration" section of https://github.com/terraform-providers/terraform-provider-aws/issues/5890 but I couldn't. I suppose the value of `TypeMap` should be primitive type (it means that we cannot use `TypeList` as the `TypeMap`'s value) by the reason reported https://github.com/hashicorp/terraform/issues/15327 .

Of course, the `cost_filter` attributes conflicts with `cost_filters`. Thus I think its good to deprecate `cost_filters` attribute.

Would you review my change and if my idea is correct?

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #5890.
Closes #10692.
Closes #12669.
Closes #13288.
Closes #19803.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Support multiple cost filter values by adding the `cost_filter` attribute and depreciating the `cost_filters` attribute
```

Output from acceptance testing:

```
[14:32:08]mozamimy@ip-10-33-164-61:terraform-provider-aws (cost-filter-multiple-values) (-'x'-).oO(
(ins)> nice -n 19 make testacc TEST=./aws TESTARGS="-run=TestAccAWSBudgetsBudget_basic" && nice -n 19 make testac
c TEST=./aws TESTARGS="-run=TestAccAWSBudgetsBudget_prefix" && nice -n 19 make testacc TEST=./aws TESTARGS="-run=
TestAccAWSBudgetsBudget_notification"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSBudgetsBudget_basic -timeout 120m
=== RUN   TestAccAWSBudgetsBudget_basic
=== PAUSE TestAccAWSBudgetsBudget_basic
=== CONT  TestAccAWSBudgetsBudget_basic
--- PASS: TestAccAWSBudgetsBudget_basic (39.27s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       39.297s
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSBudgetsBudget_prefix -timeout 120m
=== RUN   TestAccAWSBudgetsBudget_prefix
=== PAUSE TestAccAWSBudgetsBudget_prefix
=== CONT  TestAccAWSBudgetsBudget_prefix
--- PASS: TestAccAWSBudgetsBudget_prefix (34.90s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       34.924s
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSBudgetsBudget_notification -timeout 120m
=== RUN   TestAccAWSBudgetsBudget_notification
=== PAUSE TestAccAWSBudgetsBudget_notification
=== CONT  TestAccAWSBudgetsBudget_notification
--- PASS: TestAccAWSBudgetsBudget_notification (116.67s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       116.699s
```

### CODE REVIEW

**Comment by ButterflyServiceDesk** [comment]
@terraformbot can we get this in the next release please, pretty vital to be able to apply cost filters to multiple linked accounts

**Comment by JGalego** [comment]
+1

**Comment by iwt-gpoloczek** [comment]
Looking very much forward for the release of this feature! As of now, we cannot do our budget with terraform.

**Comment by aferg439** [comment]
It's been almost a year now... Can this fix be implemented, please? Thanks!

**Comment by kevingimbel** [comment]
Need this urgently! 

Multi-tag budgets are pretty powerful. The [original issue](https://github.com/terraform-providers/terraform-provider-aws/issues/5890) was opened in September 2018(!) - what is blocking this being merged and released?

**Comment by hummeltech** [comment]
Great work! What more needs to be done before this can be merged?

**Comment by yclian** [comment]
With the "post"-COVID-19 economy, we are applying budget alerts extensively to many accounts in the organization, and some aws_budgets* resources are rather coarse and would only be possible if this PR is merged.

What else do we need to make this happen?

**Comment by hummeltech** [comment]
In case this helps, I've fixed the conflicts here and rebased on `master`:

https://github.com/terraform-providers/terraform-provider-aws/compare/master...hummeltech:cost-filter-multiple-values

**Comment by yashshah** [comment]
What is blocking this change? We would love to have this released

**Comment by cparmar** [comment]
This is also causing us some issues - any news on when this could be merged?

**Comment by russ-gillespie** [comment]
I have also just run into this issue. I need to setup multiple cost_filters for data transfer usage.

**Comment by ChristophShyper** [comment]
Can someone review this PR? It will be 2 years soon...

**Comment by jfortier** [comment]
June 2021 and this is still an issue almost 2 years later! Can we get some eyes on this!

**Comment by chrono** [comment]
Is there anything blocking this? Can anything be done by the community to get this [or any change making budgets work properly] merged?

We're now adding extra budget tags, increasing report data sizes and confusing future devs working with our AWS budget simply so that this can work with the IAAS/deployment tool we're using.

**Comment by asosnovsky** [comment]
any updates on this, this ticket has been here for 2+ years and without it the budgets resource is somewhat pointless.

**Comment by elior379** [comment]
Any news about it?

We really need it !

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TESTARGS='-run=TestAccAWSBudgetsBudgetAction_\|TestAccAWSBudgetsBudget_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSBudgetsBudgetAction_\|TestAccAWSBudgetsBudget_ -timeout 180m
=== RUN   TestAccAWSBudgetsBudgetAction_basic
=== PAUSE TestAccAWSBudgetsBudgetAction_basic
=== RUN   TestAccAWSBudgetsBudgetAction_disappears
=== PAUSE TestAccAWSBudgetsBudgetAction_disappears
=== RUN   TestAccAWSBudgetsBudget_basic
=== PAUSE TestAccAWSBudgetsBudget_basic
=== RUN   TestAccAWSBudgetsBudget_Name_Generated
=== PAUSE TestAccAWSBudgetsBudget_Name_Generated
=== RUN   TestAccAWSBudgetsBudget_NamePrefix
=== PAUSE TestAccAWSBudgetsBudget_NamePrefix
=== RUN   TestAccAWSBudgetsBudget_disappears
=== PAUSE TestAccAWSBudgetsBudget_disappears
=== RUN   TestAccAWSBudgetsBudget_CostTypes
=== PAUSE TestAccAWSBudgetsBudget_CostTypes
=== RUN   TestAccAWSBudgetsBudget_Notifications
=== PAUSE TestAccAWSBudgetsBudget_Notifications
=== CONT  TestAccAWSBudgetsBudgetAction_basic
=== CONT  TestAccAWSBudgetsBudget_disappears
=== CONT  TestAccAWSBudgetsBudget_NamePrefix
=== CONT  TestAccAWSBudgetsBudget_basic
=== CONT  TestAccAWSBudgetsBudget_Notifications
=== CONT  TestAccAWSBudgetsBudget_Name_Generated
=== CONT  TestAccAWSBudgetsBudgetAction_disappears
=== CONT  TestAccAWSBudgetsBudget_CostTypes
--- PASS: TestAccAWSBudgetsBudget_disappears (16.47s)
--- PASS: TestAccAWSBudgetsBudget_NamePrefix (20.88s)
--- PASS: TestAccAWSBudgetsBudget_Name_Generated (20.94s)
--- PASS: TestAccAWSBudgetsBudget_basic (21.07s)
--- PASS: TestAccAWSBudgetsBudget_CostTypes (29.39s)
--- PASS: TestAccAWSBudgetsBudgetAction_basic (29.42s)
--- PASS: TestAccAWSBudgetsBudget_Notifications (33.87s)
--- PASS: TestAccAWSBudgetsBudgetAction_disappears (60.02s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	63.512s
```

##### GovCloud

```console
% make testacc TESTARGS='-run=TestAccAWSBudgetsBudgetAction_\|TestAccAWSBudgetsBudget_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSBudgetsBudgetAction_\|TestAccAWSBudgetsBudget_ -timeout 180m
=== RUN   TestAccAWSBudgetsBudgetAction_basic
=== PAUSE TestAccAWSBudgetsBudgetAction_basic
=== RUN   TestAccAWSBudgetsBudgetAction_disappears
=== PAUSE TestAccAWSBudgetsBudgetAction_disappears
=== RUN   TestAccAWSBudgetsBudget_basic
=== PAUSE TestAccAWSBudgetsBudget_basic
=== RUN   TestAccAWSBudgetsBudget_Name_Generated
=== PAUSE TestAccAWSBudgetsBudget_Name_Generated
=== RUN   TestAccAWSBudgetsBudget_NamePrefix
=== PAUSE TestAccAWSBudgetsBudget_NamePrefix
=== RUN   TestAccAWSBudgetsBudget_disappears
=== PAUSE TestAccAWSBudgetsBudget_disappears
=== RUN   TestAccAWSBudgetsBudget_CostTypes
=== PAUSE TestAccAWSBudgetsBudget_CostTypes
=== RUN   TestAccAWSBudgetsBudget_Notifications
=== PAUSE TestAccAWSBudgetsBudget_Notifications
=== CONT  TestAccAWSBudgetsBudgetAction_basic
=== CONT  TestAccAWSBudgetsBudget_disappears
=== CONT  TestAccAWSBudgetsBudget_Notifications
=== CONT  TestAccAWSBudgetsBudget_CostTypes
=== CONT  TestAccAWSBudgetsBudgetAction_disappears
=== CONT  TestAccAWSBudgetsBudget_Name_Generated
=== CONT  TestAccAWSBudgetsBudget_basic
=== CONT  TestAccAWSBudgetsBudget_NamePrefix
=== CONT  TestAccAWSBudgetsBudgetAction_basic
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
=== CONT  TestAccAWSBudgetsBudgetAction_disappears
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
=== CONT  TestAccAWSBudgetsBudget_disappears
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
--- SKIP: TestAccAWSBudgetsBudgetAction_basic (1.46s)
--- SKIP: TestAccAWSBudgetsBudget_disappears (1.46s)
--- SKIP: TestAccAWSBudgetsBudgetAction_disappears (1.46s)
=== CONT  TestAccAWSBudgetsBudget_basic
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
--- SKIP: TestAccAWSBudgetsBudget_basic (1.46s)
=== CONT  TestAccAWSBudgetsBudget_Name_Generated
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
--- SKIP: TestAccAWSBudgetsBudget_Name_Generated (1.46s)
=== CONT  TestAccAWSBudgetsBudget_Notifications
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
=== CONT  TestAccAWSBudgetsBudget_NamePrefix
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
--- SKIP: TestAccAWSBudgetsBudget_Notifications (1.46s)
=== CONT  TestAccAWSBudgetsBudget_CostTypes
    provider_test.go:735: skipping tests; partition aws-us-gov does not support budgets service
--- SKIP: TestAccAWSBudgetsBudget_NamePrefix (1.46s)
--- SKIP: TestAccAWSBudgetsBudget_CostTypes (1.46s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	4.404s
```

**Comment by ewbankkit** [comment]
@mozamimy Thanks for the contribution :tada: :clap:.
As we can only deprecate arguments at a major version I made changes to note that `cost_filter` is now preferred over `cost_filters` and added #20330 to deprecate `cost_filters` with **v4.0.0** which should be available later this year.

**Comment by ChristophShyper** [comment]
I'm getting 
```
╷
│ Error: Unsupported argument
│
│   on ../submodule-costs/main.tf line 127, in resource "aws_budgets_budget" "this":
│  127:   cost_filter = {
│
│ An argument named "cost_filter" is not expected here. Did you mean
│ "cost_filters"?
╵
ERRO[0052] 1 error occurred:
	* exit status 1
```

Using newest provider version
```
- Installing hashicorp/aws v3.52.0...
- Installed hashicorp/aws v3.52.0 (signed by HashiCorp)
```

**Comment by dmrzzz** [comment]
@ChristophShyper use `cost_filter {` instead of `cost_filter = {`

**Comment by ChristophShyper** [comment]
You're right! My bad, it changed type :)

**Comment by tammyllf** [comment]
cost_filter = {
           name   = "LinkedAccount"
           values = local.link_account_id
           }

Amend my cost_filter to the above, managed to run terraform init.
But hit into the following error when try to run terraform plan

```
Error: Unsupported block type

  on purple.tf line 35, in resource "aws_budgets_budget" "purple":
  35:   cost_filter {

Blocks of type "cost_filter" are not expected here.
```

I have multiple linked account, and multiple cost_filter within one tf file.
Can someone advise, what is the problem?

Thanks

**Pull request merged**
